test/v2/controllers/V2MovementsControllerSpec.scala

```diff
@@ -3043,25 +3044,88 @@ class V2MovementsControllerSpec
   "POST /movements/:movementId/messages/:messageId" - {
 
     "should return Ok when response from upscan is valid" - {
-      "and uploading to object-store succeeds" in forAll(arbitraryMovementId.arbitrary, arbitraryMessageId.arbitrary, arbitraryObjectSummaryWithMd5.arbitrary) {
-        (movementId, messageId, objectSummary) =>
-          when(
-            mockObjectStoreService.addMessage(any[String].asInstanceOf[DownloadUrl], any[String].asInstanceOf[MovementId], any[String].asInstanceOf[MessageId])(
-              any(),
-              any()
+
+      "and uploading to object-store succeeds" - {
+        "and updating the message with the object-store URL succeeds" in forAll(
+          arbitraryMovementId.arbitrary,
+          arbitraryMessageId.arbitrary,
+          arbitraryObjectSummaryWithMd5.arbitrary
+        ) {
+          (movementId, messageId, objectSummary) =>
+            when(
+              mockObjectStoreService.addMessage(
+                any[String].asInstanceOf[DownloadUrl],
+                any[String].asInstanceOf[MovementId],
+                any[String].asInstanceOf[MessageId]
+              )(
+                any(),
+                any()
+              )
+            ).thenReturn(EitherT.rightT(objectSummary))
+
+            when(
+              mockPersistenceService.updateMessage(
+                any[String].asInstanceOf[MovementId],
+                any[String].asInstanceOf[MessageId],
+                any[MessageUpdate]
+              )(
+                any(),
+                any()
+              )
+            ).thenReturn(EitherT.rightT(()))
+
+            val request = FakeRequest(
+              POST,
+              routes.V2MovementsController.attachLargeMessage(movementId, messageId).url,
+              headers = FakeHeaders(),
+              jsonSuccessUpscanResponse
             )
-          ).thenReturn(EitherT.rightT(objectSummary))
 
-          val request = FakeRequest(
-            POST,
-            routes.V2MovementsController.attachLargeMessage(movementId, messageId).url,
-            headers = FakeHeaders(),
-            jsonSuccessUpscanResponse
-          )
+            val result = sut.attachLargeMessage(movementId, messageId)(request)
 
-          val result = sut.attachLargeMessage(movementId, messageId)(request)
+            status(result) mustBe OK
+        }
+
+        "and updating the message with the object-store URL fails" in forAll(
```

Comment:
``` and updating the message with the object-store URL fails ``` this should be something related to error we receive from persistence service MessageNotFound